fdbserver/DataDistribution.actor.cpp

```diff
@@ -2569,6 +2578,186 @@ ACTOR Future<Void> waitUntilHealthy(DDTeamCollection* self, double extraDelay =
+			TraceEvent("ServerStatus", uid)
+			    .detail("Healthy", !server_status.get(uid).isUnhealthy())
+			    .detail("MachineIsValid", server_info[uid]->machine.isValid())
+			    .detail("MachineTeamSize",
+			            server_info[uid]->machine.isValid() ? server_info[uid]->machine->machineTeams.size() : -1)
+			    .detail("TeamCollectionID", tcIdString);
+			server++;
+			if (++traceEventsPrinted % 100 == 0) { // TODO: Knob-ify this number?
+				wait(yield());
+			}
+		}
+
+		i = 0;
+		TraceEvent("ServerTeamInfo").detail("Size", teams.size()).detail("TeamCollectionID", tcIdString);
+		for (; i < teams.size(); i++) {
+			const auto& team = teams[i];
+			TraceEvent("ServerTeamInfo")
+			    .detail("TeamIndex", i)
+			    .detail("Healthy", team->isHealthy())
+			    .detail("TeamSize", team->size())
+			    .detail("MemberIDs", team->getServerIDsStr())
+			    .detail("TeamCollectionID", tcIdString);
+			if (++traceEventsPrinted % 100 == 0) { // TODO: Knob-ify this number?
+				wait(yield());
+			}
+		}
+
+		i = 0;
+		TraceEvent("MachineInfo").detail("Size", machine_info.size()).detail("TeamCollectionID", tcIdString);
+		state std::map<Standalone<StringRef>, Reference<TCMachineInfo>>::iterator machine = machine_info.begin();
+		state bool isMachineHealthy = false;
+		for (; i < machine_info.size(); i++) {
+			Reference<TCMachineInfo> _machine = machine->second;
+			if (!_machine.isValid() || machine_info.find(_machine->machineID) == machine_info.end() ||
+			    _machine->serversOnMachine.empty()) {
+				isMachineHealthy = false;
+			}
+
+			// Healthy machine has at least one healthy server
+			for (auto& server : _machine->serversOnMachine) {
+				if (!server_status.get(server->id).isUnhealthy()) {
+					isMachineHealthy = true;
+				}
+			}
+
+			isMachineHealthy = false;
+			TraceEvent("MachineInfo")
+			    .detail("MachineInfoIndex", i)
+			    .detail("Healthy", isMachineHealthy)
+			    .detail("MachineID", machine->first.contents().toString())
+			    .detail("MachineTeamOwned", machine->second->machineTeams.size())
+			    .detail("ServerNumOnMachine", machine->second->serversOnMachine.size())
+			    .detail("ServersID", machine->second->getServersIDStr())
+			    .detail("TeamCollectionID", tcIdString);
+			machine++;
+			if (++traceEventsPrinted % 100 == 0) { // TODO: Knob-ify this number?
+				wait(yield());
+			}
+		}
+
+		i = 0;
+		TraceEvent("MachineTeamInfo").detail("Size", machineTeams.size()).detail("TeamCollectionID", tcIdString);
+		for (; i < machineTeams.size(); i++) {
+			const auto& team = machineTeams[i];
+			TraceEvent("MachineTeamInfo")
+			    .detail("TeamIndex", i)
+			    .detail("MachineIDs", team->getMachineIDsStr())
+			    .detail("ServerTeams", team->serverTeams.size())
+			    .detail("TeamCollectionID", tcIdString);
+			if (++traceEventsPrinted % 100 == 0) { // TODO: Knob-ify this number?
+				wait(yield());
+			}
+		}
+
+		i = 0;
+		TraceEvent("LocalityRecordKeyName")
```

Comment:
We can comment out this `LocalityRecordKeyName ` or knob-ify it.
The locality info is not as useful as the above server and machine status info. 

In the first version, let's don't print the LocalityRecordKeyName and  MachineLocalityMap info.

Reply:
Done. Removed.